### ISSUE TITLE
Change packaging to separate core and platform specific files to different artifacts.

### ISSUE THREAD

**Issue by hvesalai** [bytedeco/javacpp-presets#262]
Currently using this project requires that two artifacts with the same id be included on the classpath. This is not the way it should be and causes problems.

At the moment, using the project requires that, for example, both of these artifacts are on the classpath

```
org.bytedeco.javacpp-presets:opencv:3.1.0-1.2
org.bytedeco.javacpp-presets:opencv:3.1.0-1.2:linux-x86_64
```

I.e. there are two modules with the exact same artifact id. In our case this results in sbt generating an ivy file with both artifacts in one dependency:

```
<dependency org="org.bytedeco.javacpp-presets" name="opencv" rev="3.1.0-1.2" conf="compile->default(compile)">
  <artifact name="opencv" type="jar" ext="jar" conf="compile"/>
  <artifact name="opencv" type="jar" ext="jar" conf="*" e:classifier="linux-x86_64"/>
</dependency>
```

And so, while SBT it self downloads both artifacts, the downstream projects only download the first one (since in their interpretation there is just two alternatives of which they choose one).

You can contrast and compare this with how build systems handle cases where multiple artifacts with the same artifact id, but different version are resolved. Most build systems don't include both versions to the classpath (that would cause a ton of problems), but rather choose the latest one or require the developer to choose. In the same way this project is now requiring there two modules with the same artifact id (and same version), but with different classifier.

The proper way to package and publish opencv would be to have the common java libraries be in one artifact and the platform specific stuff in an other. I.e. `org.bytedeco.javacpp-presets:opencv-common:3.1.0-1.2` would be for the java classes and then `org.bytedeco.javacpp-presets:opencv:3.1.0-1.2:platform` (where platform would be e.g. `linux-x86_64`) would be for the native libraries. The later would depend on the former so including `org.bytedeco.javacpp-presets:opencv:3.1.0-1.2:linux-x86_64` would result in both the linux libraries and the java classes in org.bytedeco.javacpp-presets:opencv-common:3.1.0-1.2 to be included on to the classpath.

**Comment by hvesalai** [comment]
Note: I'm not saying that the current packaging wouldn't be correct in maven terms, but rather that it is causing problems with some non-maven dependency management systems and that there is a simple fix which would allow all build systems to use this project without problems.

**Comment by saudet** [comment]
It's not a simple fix. If you are serious about fixing this, please make a prototype and show that everything builds and functions properly on all platforms for all build tools.

In any case, check out my own solution in the `platform` branch. That should take care of all these issues, but obviously I need people to test it out.

**Comment by hvesalai** [comment]
Do you have SNAPSHOT releases of the platform branch available in some repository? I can test it for linux on sbt and sbt-launcher.

**Comment by saudet** [comment]
No, but you could try ND4J which uses the same mechanism: http://nd4j.org/getstarted.html

And there's a sample build.sbt file as part of ND4S: https://github.com/deeplearning4j/nd4s/

Let me know if you encounter any problems with that! Thanks

**Comment by saudet** [comment]
"Platform artifacts" have been introduced in the newly released version 1.3, and they work fine with sbt. Enjoy!

**Comment by hvesalai** [comment]
Nice!
